towerlib/towerlib.py

```diff
@@ -1997,24 +2034,707 @@ def schedules(self):
+        if not response.ok:
+            self._logger.error("Error getting the job events. response was: {})".format(response.text))
+            return None
+        else:
+            return response.json().get('results', [])
+
+    def get_job_dates_by_host(self, host):
+        """Get job dates from ansible tower for a given host id when the job event was created.
+
+        Args:
+            host: the host object of tower instance.
+
+        Returns:
+            list: a list containing the formatted datetime objects.
+
+        """
+        self._logger.debug("Start getting job events for the host '{}'".format(host.name))
+        job_events = self.get_job_events_by_host(host)
+        self._logger.debug("Finished getting job events for the host '{}'".format(host.name))
+        if job_events is None:
+            self._logger.error("No job events found.")
+            return None
+        else:
+            self._logger.debug("job events found")
+            job_dates = []
+            for job_event in job_events:
+                job_event_time = job_event['created']
+                job_dates.append(job_event_time)
+            return job_dates
+
+    def get_groups_by_host(self, host):
+        """Get groups for a particular host, which are directly connected.
+
+        Args:
+            host: the host object.
+
+        Returns:
+            list: list of custom groups.
+
+        """
+        url = "{api}/hosts/{id}/groups/".format(api=self.api, id=host.id)
+        response = self.session.get(url)
+        if not response.ok:
+            self._logger.error("Error getting the project updates. response was: {})".format(response.text))
+            return None
+        else:
+            results = response.json().get('results', [])
+            groups = []
+            for group in results:
+                groups.append(group)
+            return groups
+
+    def get_all_groups_by_host(self, host):
+        """Get groups for a particular host, which are directly and indirectly connected.
+
+        Args:
+            host: the host object.
+
+        Returns:
+            list: list of custom groups.
+
+        """
+        url = "{api}/hosts/{id}/all_groups/".format(api=self.api, id=host.id)
+        response = self.session.get(url)
+        if not response.ok:
+            self._logger.error("Error getting the project updates. response was: {})".format(response.text))
+            return None
+        else:
+            results = response.json().get('results', [])
+            groups = []
+            for group in results:
+                groups.append(group)
+            return groups
+
+    def get_all_groups(self):
+        """Get all the groups of the given ansible tower.
+
+        Returns:
+            list: list of all the groups as EntityManager object.
+
+        """
+        groups = [item for item in self.groups]
+        return groups
+
+
+    def disable_scm_update_on_launch_for_all_projects(self):
```

Comment:
This can be potentially quite disruptive as it will reflect on all projects of a tower deployment. I would argue a bit against such functionality which can already be achieved by iterating over all projects and setting scm_update_on_launch to False.